tools/cert-sidecar/app/CertificatesMgmt.js

```diff
@@ -0,0 +1,289 @@
+      logger.debug(`retrieveCRL: New Intention CRL ${crl}`);
+      logger.debug(`retrieveCRL: Old Fingerprint CRL ${this.crlFingerprint}`);
+      logger.debug(`retrieveCRL: New Intention Fingerprint CRL ${newIntentionFingerprintCRL}`);
+
+      if (!isFirstRetrieveTime) {
+        checkIfIsADiffCRL = newIntentionFingerprintCRL
+          !== this.crlFingerprint;
+      }
+
+      logger.debug(`retrieveCRL: Is this the First Retrieve? ${isFirstRetrieveTime}`);
+      logger.debug(`retrieveCRL: Are the CRL differents? ${checkIfIsADiffCRL}`);
+
+      if (isFirstRetrieveTime || checkIfIsADiffCRL) {
+        logger.info('retrieveCRL: Saving new CRL');
+
+        this.crl = crl;
+        this.crlFingerprint = newIntentionFingerprintCRL;
+
+        await createFile(this.pathCrl, this.crl);
+      } else {
+        logger.info('retrieveCRL: CRL doesn\'t change');
+      }
+    } catch (e) {
+      logger.error('retrieveCRL:', e);
+      throw e;
+    }
+    logger.info('retrieveCRL: ...ending retrieve CRL.');
+  }
+
+  /**
+   * Generate private key and create a new file in 'certs.files.key'
+   */
+  async generatePrivateKey() {
+    logger.info('generatePrivateKey: Generating  a Private Key...');
+    try {
+      const privateKey = await X509Utils.generatePrivateKey();
+      this.privateKey = privateKey;
+      await createFile(this.pathKey, this.privateKey);
+    } catch (e) {
+      logger.error('generatePrivateKey:', e);
+      throw e;
+    }
+    logger.info('generatePrivateKey: ...ending generate a Private Key.');
+  }
+
+  /**
+   * Generate certificate and create a new file in 'certs.files.cert'
+   */
+  async generateCertificate() {
+    logger.info('generateCertificate: Generating a Certificate...');
+    try {
+      const csr = await X509Utils
+        .generateCsr(
+          this.privateKey,
+          configCerts['common.name'],
+          configCerts.hostnames,
+        );
+      const cert = await x509Req.createCertificateByCSR(csr);
+      this.cert = cert;
+      await createFile(this.pathCert, this.cert);
+    } catch (e) {
+      logger.error('generateCertificate:', e);
+      throw e;
+    }
+    logger.info('generateCertificate: ... ending generate a Certificate.');
+  }
+
+  /**
+   * Retrieve ca certificate and create a new file in 'certs.files.ca'
+   */
+  async retrieveCaCert() {
+    logger.info('retrieveCaCert: Retrieving  a CA Certificate...');
+    try {
+      const ca = await x509Req.getCACertificate();
+      this.ca = ca;
+      await createFile(this.pathCA, this.ca);
+    } catch (e) {
+      logger.error('retrieveCaCert:', e);
+      throw e;
+    }
+    logger.info('retrieveCaCert: ... ending retrieve a CA Certificate...');
+  }
+
+  /**
+   * The toString() method returns a string representing this class.
+   *
+   *  @return {string } a string representing this class
+   */
+  toString() {
+    return JSON.stringify({
+      privateKey: this.privateKey,
```

Comment:
A point of attention here, for security, the private key cannot appear in any log.
My suggestion is not to display it in `toString()`